src/control/server/storage/scm/ipmctl.go

```diff
@@ -59,8 +59,8 @@ func (sv semVer) String() string {
 	return strings.Join(sv, ".")
 }
 
-func validateIpmctlVer(log logging.Logger, sv semVer) error {
-	for _, badVer := range badIpmctlVers {
+func validateSemVer(sv semVer, badList []semVer) error {
```

Comment:
It's a minor quibble, but what I was trying to convey is that this function should probably be less focused on this one use case and more focused on the general problem of checking whether or not a given semVer is present in an arbitrary list of semVers. Yes, it works now, for this one problem. But with minor adjustments you get a more general solution that can be easily repurposed without having to rewrite anything.

Reply:
`validateSemVer(sv semVer, badList []semVer)` I don't understand how this is not "focused on the general problem of checking whether or not a given semVer is present in an arbitrary list of semVers", that seems to be exactly what the function is doing and signature seems to indicate exactly that. let me know how you think it could be improved.

Reply:
addressed in most recent commit